inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_binary_elementwise.cpp

```diff
@@ -48,7 +48,25 @@ void processBinaryEltwise(std::shared_ptr<ngraph::Node> eltwise, size_t lhsIndex
         broadcastInput = std::make_shared<ngraph::opset3::Concat>(ngraph::OutputVector{broadcastConst, broadcastInput}, 0);
     }
 
-    const auto shape = std::make_shared<ngraph::opset3::Maximum>(lhsInput, rhsInput);
+    const auto& lhsInputShape = lhsInput.get_partial_shape();
+    const auto& rhsInputShape = rhsInput.get_partial_shape();
+
+    VPU_THROW_UNLESS(lhsInputShape.is_static() && lhsInputShape.rank().get_length() == 1,
+        "DynamicToStaticShape transformation for {} of type {} expects lhs input shape to be static 1D vector, actual is {}",
+        eltwise->get_friendly_name(), eltwise->get_type_info(), lhsInputShape);
+
+    VPU_THROW_UNLESS(rhsInputShape.is_static() && rhsInputShape.rank().get_length() == 1,
+        "DynamicToStaticShape transformation for {} of type {} expects rhs input shape to be static 1D vector, actual is {}",
+        eltwise->get_friendly_name(), eltwise->get_type_info(), rhsInputShape);
+
+    const auto& lhsMin = std::make_shared<ngraph::opset5::ReduceMin>(lhsInput, ngraph::opset5::Constant::create(shapeElementType, {}, {0}));
```

Comment:
Why do you need this `ReduceMin`? It looks like we will get the output shape full of zeros in case if we have at least one zero in one of input shapes. Is it expected?

Reply:
If we have at least one zero in any of input shapes it means we have an empty input, right?
If we have an empty input it means second input must broadcast to empty shape (discussed with @slyalin and @lazarevevgeny)

Regarding shape full of zeros: it's a good question which shape exactly must output have in empty input scenario. In the sake of simplicity I preserver only rank and fill all dimensions with zero. If anyone has some concerns about that I'm happy to discuss rationale and possible implementation.

Reply:
> Why do you need this ReduceMin?

I wanted to use `ReduceProd` at first, but this operation is not support in Myriad-X plugin, so used `ReduceMin` assuming dimensions are non-negative.

Reply:
I think we need to do this based on the broadcast specification.
In case we have something like [100, 1, 0, 50] and [1], I expect to see [100, 1, 0, 50] as the output shape, because if some of the dimensions is broadcasting to zero, it doesn't mean that the rule changes for other dimensions.
Also some of these output dimensions may be extracted via `ShapeOf` afterwards, so it may affect some cases on inference.

Reply:
Please, take a look at my solution once more